fluster/decoders/ffmpeg.py

```diff
@@ -119,38 +72,111 @@ def decode(
+
+        # Passthrough timestamp from the demuxer to the muxer
+        if self.ffmpeg_version and self.ffmpeg_version >= (5, 1):
+            command.extend(["-fps_mode", "passthrough"])
+        else:
+            command.extend(["-vsync", "passthrough"])
+
+        # Hardware download
+        download = ""
+        if self.hw_acceleration and self.hw_download:
+            if output_format not in self.hw_download_mapping:
+                raise Exception(
+                    f"No matching ffmpeg pixel format found for {output_format}"
+                )
+            download = f"hwdownload,format={self.hw_download_mapping[output_format]},"
+
+        # Output format filter
+        command.extend(["-filter", f"{download}format=pix_fmts={output_format.value}"])
+
+        # MD5 muxer
+        if self.use_md5_muxer and not keep_files:
+            command.extend(["-f", "md5", "-"])
+            output = run_command_with_output(command, timeout=timeout, verbose=verbose)
+            md5sum = re.search(r"MD5=([0-9a-fA-F]+)\s*", output)
+            if not md5sum:
+                raise Exception("No MD5 found in the program trace.")
+            return md5sum.group(1).lower()
+
+        # Output file
+        command.extend(["-f", "rawvideo", output_filepath])
+        run_command(command, timeout=timeout, verbose=verbose)
         return file_checksum(output_filepath)
 
     @lru_cache(maxsize=128)
     def check(self, verbose: bool) -> bool:
         """Checks whether the decoder can be run"""
-        # pylint: disable=broad-except
-        if self.hw_acceleration:
-            try:
-                command = None
-
-                if self.wrapper:
-                    command = [self.binary, "-decoders"]
-                else:
-                    command = [self.binary, "-hwaccels"]
-
-                output = subprocess.check_output(
-                    command, stderr=subprocess.DEVNULL
-                ).decode("utf-8")
-                if verbose:
-                    print(f'{" ".join(command)}\n{output}')
-
-                if self.wrapper:
-                    return self.api.lower() in output
-
-                return f"{os.linesep}{self.api.lower()}{os.linesep}" in output
-            except Exception:
-                return False
-        else:
-            return super().check(verbose)
+        if not super().check(verbose):
+            return False
+
+        # Check if codec is supported
+        codec_mapping = {
+            Codec.H264: "h264",
+            Codec.H265: "hevc",
+            Codec.VP8: "vp8",
+            Codec.VP9: "vp9",
+            Codec.AV1: "av1",
+        }
+        if self.codec not in codec_mapping:
+            return False
+        self.ffmpeg_codec = codec_mapping[self.codec]
+
+        # Get ffmpeg version
+        output = _run_ffmpeg_command(self.binary, "-version", verbose=verbose)
+        version = re.search(r"ffmpeg version (\d+)\.(\d+)(?:\.(\d+))?", output)
```

Comment:
this re doesn't match when ffmpeg is compiled from git (N-git-describe)

Reply:
I have adjusted the pattern to also match builds from `release/` branches (`n`-prefix) and changed the version check to presume that `-fps_mode` can be used when no version can be determined.